website/ref/language.md

```diff
@@ -102,27 +104,79 @@ metacharacters; use quoted strings instead. For instance, to echo a star, write
 in line continuations; their use elsewhere is reserved will cause a syntax
 error.
 
-## Notes
+### Notes
 
 The three syntaxes above all evaluate to strings, and they are interchangeable.
 For instance, `xyz`, `'xyz'` and `"xyz"` are different syntaxes for the same
-string, and they are always equivalent.
+string, and they are always equivalent with the exception of
+**escape sequences** as documented above.
+
+## Number
+
+Elvish has a numeric data type, `float64`, created using this syntax:
+`(float64 $number)`. For convenience, all Elvish functions that
+expect a numeric argument will accept a string that can be parsed as a
+number. They will automatically convert the string to a `float64`. Which
+means explicit `float64` constructors are almost never needed. Not even
+for special values like infinity (`Inf`) or not-a-number (`NaN`). Note
+that the letter case of these magic values doesn't matter so `inf` and
+`Inf` are equivalent.
+
+The `$number` argument can be a string in the following formats:
+
+* decimal notation (e.g., `10`),
+* hexadecimal notation (e.g., `0xA`),
+* octal notation (e.g., `0o12`),
+* binary notation (e.g., `0b1010`),
+* floating point notation (e.g., `10.0`), and
+* scientific notation (e.g., `1.0e1`).
+
+Note that the examples in the above list all express the same value.
```

Comment:
Better say this before the bullet, instead of after the bullet. Say

The `$number` argument can be a string in any of the following formats (the examples all express the same value):